### PR TITLE
Files of SCF Assignment 7 - Sparse Matrix

### PR DESCRIPTION
**Pull request by meta-chirag-jain**

### CODE REVIEW

File: com/metacube/SCF/Assignment7/SparseMatrix.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+public final class SparseMatrix {
+
+	private int[][] sparseMatrix;
+	private int rows;
+	private int columns;
+	
+	private static final int VALUE = 0;
+    private static final int ROW_NUMBER = 1;
+    private static final int COL_NUMBER = 2;
+
```
Comment (pallavisoni) [inline com/metacube/SCF/Assignment7/SparseMatrix.java]: correct the formatting.

File: com/metacube/SCF/Assignment7/SparseMatrix.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	private int[][] sparseMatrix;
+	private int rows;
+	private int columns;
+	
+	private static final int VALUE = 0;
+    private static final int ROW_NUMBER = 1;
+    private static final int COL_NUMBER = 2;
+	
+	public SparseMatrix(int[][] givenArray) {
+		int length = 0;
```
Comment (pallavisoni) [inline com/metacube/SCF/Assignment7/SparseMatrix.java]: Add method level comment.

File: com/metacube/SCF/Assignment7/SparseMatrix.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		int maxColLength = 0;														//if user gives different column size in different rows
+		
+		if(givenArray == null) {
+			throw new AssertionError("Cant be null.");
+		}
+
```
Comment (pallavisoni) [inline com/metacube/SCF/Assignment7/SparseMatrix.java]: Also check for empty array.

Comment (meta-chirag-jain) [reply com/metacube/SCF/Assignment7/SparseMatrix.java]: done

File: com/metacube/SCF/Assignment7/SparseMatrix.java (hunk 2 above)
Comment (meta-chirag-jain) [reply com/metacube/SCF/Assignment7/SparseMatrix.java]: done

File: com/metacube/SCF/Assignment7/SparseMatrix.java (hunk 1 above)
Comment (meta-chirag-jain) [reply com/metacube/SCF/Assignment7/SparseMatrix.java]: done

**Pull request merged**